include/tm/vector.hpp

```diff
@@ -4,17 +4,16 @@
 #include <initializer_list>
 #include <stddef.h>
 
-#define TM_MIN(a, b) (a < b ? a : b)
-
 namespace TM {
 
 const int VECTOR_GROW_FACTOR = 2;
 const int VECTOR_MIN_CAPACITY = 10;
 
+#define ARRAY_OF_SIZE(size) reinterpret_cast<T *>(new unsigned char[size * sizeof(T)] {})
+
```

Comment:
I'm still a C++ noob, but I think you can do this without a macro using `constexpr`...

```
    constexpr T *array_of_size(size_t size) {
        return reinterpret_cast<T *>(new unsigned char[size * sizeof(T)] {});
    }
```

Reply:
I am not sure you can, wouldn't a constexpr be evaluated at compile-time (maybe we understood this feature differently)? I think you could do this with a function that is always inlined, but it's quite a minimal macro and unlikely to be changed :) Plus just to be safe I undefine it at the end because I don't want it to be used anywhere else :D

Reply:
OK fair enough, carry on.